src/VecSim/algorithms/hnsw/hnsw_batch_iterator.cpp

```diff
@@ -126,6 +124,7 @@ candidatesMaxHeap HNSW_BatchIterator::scanGraph(candidatesMinHeap &candidates,
             float candidate_dist = dist_func(this->getQueryBlob(), (const void *)candidate_data,
                                              this->space->get_data_dim());
             candidates.emplace(candidate_dist, candidate_id);
+            __builtin_prefetch(hnsw_index->get_linklist_at_level(candidates.top().second, 0));
```

Comment:
Here we prefetch each time we update the candidates set. Consider benchmarking an alternative where we prefetch once before calling `processCandidates`

Reply:
also, if you only prefetch layer 0, you can call directly to `get_linklist0`

Reply:
In `addPpoint` we might prefetch other layers as well...